api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1190,6 +1205,10 @@ message Cluster {
   // If ``connection_pool_per_downstream_connection`` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // b/216628756
+  // Add configurations for happy eyeballs algorithm
+  HappyEyeballsConfig happy_eyeballs_config = 57;
```

Comment:
Maybe this should go inside `UpstreamConnectionOptions` instead of being a top-level field in the `Cluster` proto?